File: stack/1_stack.js
class Stack {
    constructor() {
        this.stack = [];
    }

    is_empty() {
        if (this.stack.length == 0) {
            return true;
        }
        return false;
    }

    push(item) {
        this.stack.push(item);
    }

    pop() {
        if (this.is_empty()){
            return "Stack IS Empty";
        }
        return this.stack.pop();
    }

    peek(){
        if (this.is_empty()){
            return "Stack IS Empty";
        }
        return this.stack[this.stack.length - 1];
    }

    size(){
        return this.stack.length;
    }


};



const s = new Stack();
s.push(10);
s.push(20);
console.log(s.peek());  // 20
console.log(s.pop());   // 20
console.log(s.is_empty());  // false

File: 4_stack/1_stack.py
class Stack:
    def __init__(self):
        self.stack = []
    
    def is_empty(self):
        if not self.stack:
            return True
        return False
    
    def push(self, item):
        self.stack.append(item)
    
    def pop(self):
        if self.is_empty():
            return "Stack is empty"
        return self.stack.pop() # item on the top

    def peek(self):
        if self.is_empty():
            return "Stack is empty"
        return self.stack[-1]
    
    def size(self):
        return len(self.stack)

s = Stack()
s.push(10)
s.push(20)
s.pop()

print(s.peek())
print(s.is_empty())
